src/modding/FullModDetails.cs

```diff
@@ -1,22 +1,54 @@
 ﻿using System;
+using System.Collections.Generic;
+using Godot;
+using Newtonsoft.Json;
 
 /// <summary>
 ///   An extended version of <see cref="ModInfo"/> that contains non-mod author controlled data
 /// </summary>
 public class FullModDetails : IEquatable<FullModDetails>
 {
+    [JsonConstructor]
     public FullModDetails(string internalName, string folder, ModInfo info)
     {
         InternalName = internalName;
         Folder = folder;
         Info = info;
     }
 
+    public FullModDetails(string internalName)
+    {
+        InternalName = internalName;
+        Info = new ModInfo();
+        Folder = string.Empty;
+    }
+
     public string InternalName { get; }
     public string Folder { get; set; }
 
+    /// <summary>
+    ///   Is the mod compatible with the current version of thrive?
+    /// </summary>
+
+    // 1 = Compatible, 0 = Unknown, -1 = Might Not Be Compatible, -2 = Not Compatible
+    public int IsCompatibleVersion { get; set; }
```

Comment:
Also, I like the idea of having a richer compatibility system. However, it the use of an `int` any better than a hardcoded `string` (as an internal name that we'll be able to translate later?). Real question there, I'm not pushing for it either.

Reply:
It does allow for easier filtering of only known compatible mods by only allowing positive integers and vice-versa, but would it be better as a enum?

Reply:
It would be a lot better as an enum. You can create an extension method for the enum which checks if it is compatible. So you can write code line `enumValue.IsCompatible()`.

Reply:
I agree, an enum would be much nicer.